tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -267,7 +271,31 @@ private static void AddAppsecSpecificInstrumentations()
         public void Dispose()
         {
             _waf?.Dispose();
-            _rateLimiter.Dispose();
+            _cancellationTokenSource?.Cancel();
+            _cancellationTokenSource?.Dispose();
+        }
+
+        private async void RunRateLimitLoop(TimeSpan period, CancellationToken cancellationToken)
```

Comment:
`async void ` is a pattern to avoid (cf https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void)

Reply:
I dunno, I avoid `async void` [like the plague](https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void).

Reply:
I like this [SO post](https://stackoverflow.com/a/12804036) as well. Quoting: 

> I know some people call it "fire and forget" but I usually use the phrase "fire and crash".

Reply:
It's gone now and I'm certainly convinced a `Timer` instance is a better choice than using `Task` here 😜